core/server/web/shared/middlewares/serve-public-file.js

```diff
@@ -4,47 +4,59 @@ const path = require('path');
 const urlService = require('../../../services/url');
 
-// ### servePublicFile Middleware
-// Handles requests to robots.txt and favicon.ico (and caches them)
-function servePublicFile(file, type, maxAge) {
+function createPublicFileMiddleware(file, type, maxAge) {
     let content;
     const publicFilePath = config.get('paths').publicFilePath;
     const filePath = file.match(/^public/) ? path.join(publicFilePath, file.replace(/^public/, '')) : path.join(publicFilePath, file);
     const blogRegex = /(\{\{blog-url\}\})/g;
     const apiRegex = /(\{\{api-url\}\})/g;
 
     return function servePublicFile(req, res, next) {
-        if (req.path === '/' + file) {
-            if (content) {
-                res.writeHead(200, content.headers);
-                res.end(content.body);
-            } else {
-                fs.readFile(filePath, (err, buf) => {
-                    if (err) {
-                        return next(err);
-                    }
-
-                    if (type === 'text/xsl' || type === 'text/plain' || type === 'application/javascript') {
-                        buf = buf.toString().replace(blogRegex, urlService.utils.urlFor('home', true).replace(/\/$/, ''));
-                        buf = buf.toString().replace(apiRegex, urlService.utils.urlFor('api', {cors: true, version: 'v0.1', versionType: 'content'}, true));
-                    }
-                    content = {
-                        headers: {
-                            'Content-Type': type,
-                            'Content-Length': buf.length,
-                            ETag: `"${crypto.createHash('md5').update(buf, 'utf8').digest('hex')}"`,
-                            'Cache-Control': `public, max-age=${maxAge}`
-                        },
-                        body: buf
-                    };
-                    res.writeHead(200, content.headers);
-                    res.end(content.body);
-                });
+        if (content) {
+            res.writeHead(200, content.headers);
+            return res.end(content.body);
+        }
+        fs.readFile(filePath, (err, buf) => {
+            if (err) {
+                return next(err);
             }
+
+            let str = buf.toString();
+
+            if (type === 'text/xsl' || type === 'text/plain' || type === 'application/javascript') {
+                str = str.replace(blogRegex, urlService.utils.urlFor('home', true).replace(/\/$/, ''));
+                str = str.replace(apiRegex, urlService.utils.urlFor('api', {cors: true, version: 'v0.1', versionType: 'content'}, true));
+            }
+
+            content = {
+                headers: {
+                    'Content-Type': type,
+                    'Content-Length': Buffer.from(str).length,
+                    ETag: `"${crypto.createHash('md5').update(str, 'utf8').digest('hex')}"`,
+                    'Cache-Control': `public, max-age=${maxAge}`
+                },
+                body: str
```

Comment:
This will always pass `buf.toString()` in body here compared to possibility of just `buf` being passed in the previous code, I think, probably unintended change?

Reply:
Nice eye, though as we pass it to `res.end` I don't think it matters whether we pass buffer or string, it will get chunked and sent to the client the same either way.